components/turret.py

```diff
@@ -126,24 +126,31 @@ def setup(self) -> None:
         self.azimuth_history = deque(maxlen=self.MEMORY_CONSTANT)
 
     # Slew to the given absolute angle in radians in the robot coordinate system.
-    # We assume that the angle is between -pi and pi
     # If no index has been seen yet, we have no reference for the angle, so we
     # ignore the command.
     #
     # TODO: perhaps we should return an error so the calling code can know?
     def slew_to_azimuth(self, angle: float) -> None:
-        angle -= math.pi  # This converts to turret coordinate system
-        # Now we normalise to +- pi
-        while angle < -math.pi:
-            angle += math.tau
-        while angle > math.pi:
-            angle -= math.tau
         if self.index_found:
             self.current_state = self.SLEWING
-            self.motor._slew_to_counts(int(angle * self.COUNTS_PER_TURRET_RADIAN))
+            turret_angle = self.robot_to_turret(angle)
+            self.motor._slew_to_counts(
+                int(turret_angle * self.COUNTS_PER_TURRET_RADIAN)
+            )
         else:
             print("slew_to_azimuth() called before index found")
 
+    # Convert an angle in the robot coordinate system to the turret coordinate
+    # system, which is rotated by 180 degrees (i.e. 0 is backwards on the robot).
+    def robot_to_turret(self, angle: float) -> float:
```

Comment:
Given that this function doesn't actually use self, should it be a method? (We could make it a staticmethod I guess.)

Reply:
Done.